atef/ui/action_row_widget.ui

```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <class>Form</class>
+ <widget class="QWidget" name="Form">
+  <property name="geometry">
+   <rect>
+    <x>0</x>
+    <y>0</y>
+    <width>598</width>
+    <height>45</height>
+   </rect>
+  </property>
+  <property name="windowTitle">
+   <string>Form</string>
+  </property>
+  <layout class="QHBoxLayout" name="horizontalLayout">
+   <item>
+    <widget class="QToolButton" name="child_button">
+     <property name="text">
+      <string>...</string>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <widget class="QLineEdit" name="name_edit">
+     <property name="sizePolicy">
+      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
+       <horstretch>0</horstretch>
+       <verstretch>0</verstretch>
+      </sizepolicy>
+     </property>
+     <property name="placeholderText">
+      <string>action name</string>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <widget class="QToolButton" name="target_button">
+     <property name="text">
+      <string>select a target</string>
+     </property>
+     <property name="popupMode">
+      <enum>QToolButton::InstantPopup</enum>
+     </property>
+     <property name="toolButtonStyle">
+      <enum>Qt::ToolButtonTextOnly</enum>
+     </property>
+     <property name="arrowType">
+      <enum>Qt::DownArrow</enum>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <widget class="QToolButton" name="setting_button">
```

Comment:
I really like these widgets! I noticed a couple things while playing around with them:
1. It's possible to have the 'happi item' selector window go in the background. If you then click 'select a target' again, it won't bring that window to the top. If that window is entirely behind the main atef, then it gives the impression that it's broken
2. If I double-click the component name in the happi item window (or even select it and click the 'Select' button):
<img width="325" alt="image" src="https://user-images.githubusercontent.com/5139267/232635787-f721759c-ef59-49e6-9537-92c718bb1474.png">
The bottom pop-up goes away and the component setting is unchanged. I have to click it again to show the bottom window and then press 'Apply'. I think that step should be unnecessary here
3. 'Confirm' value has no text?
<img width="304" alt="image" src="https://user-images.githubusercontent.com/5139267/232635969-9f4c874e-2a18-41ac-b607-62431ab7c399.png">

Reply:
1: Will be addressed!
2: I actually struggled with this for a long time and couldn't come up with a solution.  I can try again, but my recollection is that changing the default behavior of these QToolButtons required invasive surgery
3: in our rhel fusion style these button boxes always come with icons, which is why I removed the text.   Once we unify the styles (#152)  this should hopefully no longer an issue

Reply:
Hmm, I'd be happy to help look into (2). One of the goals of those component-selector widgets was really to be reusable - such that all you have to do is hook into signals to get the information you want. If that's not how it is, then it needs to be fixed

Reply:
This wasn't as bad as I remembered it being 🙄